src/binding/transaction_log_file.h

```diff
@@ -125,6 +138,21 @@ struct TransactionLogFile final {
 	 */
 	void writeEntries(TransactionLogEntryBatch& batch, const uint32_t maxFileSize = 0);
 
+	/**
+	 * Return a memory map of the file and mark it as in use
+	 */
+	MemoryMap* getMemoryMap(uint32_t fileSize);
```

Comment:
I really think this should return a `std::shared_ptr<MemoryMap>`. This pointer is leaving the safety of the `TransactionLogFile` instance, passed through the store to `TransactionLog::GetMemoryMapOfFile()`.